demos/common/python/openvino/model_zoo/model_api/models/instance_segmentation.py

```diff
@@ -0,0 +1,293 @@
+"""
+ Copyright (c) 2020 Intel Corporation
+
+ Licensed under the Apache License, Version 2.0 (the "License");
+ you may not use this file except in compliance with the License.
+ You may obtain a copy of the License at
+
+      http://www.apache.org/licenses/LICENSE-2.0
+
+ Unless required by applicable law or agreed to in writing, software
+ distributed under the License is distributed on an "AS IS" BASIS,
+ WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ See the License for the specific language governing permissions and
+ limitations under the License.
+"""
+
+import cv2
+import numpy as np
+
+from .image_model import ImageModel
+from .utils import RESIZE_TYPES, load_labels, nms, pad_image
+
+
+class MaskRCNNModel(ImageModel):
+    def __init__(self, model_adapter, prob_threshold=0.5, labels=None, keep_aspect_ratio=False):
+        super().__init__(model_adapter)
+        self.resize_type = 'fit_to_window' if keep_aspect_ratio else 'standard'
+        self.resize = RESIZE_TYPES[self.resize_type]
+        self.prob_threshold = prob_threshold
+        self._check_io_number((1, 2), (3, 4, 5, 8))
+        self.type = 'segmentoly' if len(self.inputs) == 2 else 'mask_rcnn'
+        if isinstance(labels, (list, tuple)):
+            self.labels = labels
+        else:
+            self.labels = load_labels(labels) if labels else None
+
+        self.output_blob_name = self._get_outputs()
+
+    def _get_outputs(self):
+        if self.type == 'segmentoly':
+            return self._get_segmentoly_outputs()
+        outputs = {}
+        for layer_name in self.outputs:
+            if layer_name.startswith('TopK'):
+                continue
+            layer_shape = self.outputs[layer_name].shape
+
+            if len(layer_shape) == 1:
+                outputs['labels'] = layer_name
+            elif len(layer_shape) == 2:
+                outputs['boxes'] = layer_name
+            elif len(layer_shape) == 3:
+                outputs['masks'] = layer_name
+            else:
+                raise Exception("Unexpected output layer shape {} with name {}".format(layer_shape, layer_name))
```

Comment:
Please, use a specific error for model API wrappers, which is defined by `WrapperError` class.

Reply:
Fixed.